.github/workflows/go.yml

```diff
@@ -14,16 +14,16 @@ jobs:
         with:
           go-version: "^1.19.0"
       - name: Run golangci-lint
-        uses: golangci/golangci-lint-action@v3.2.0
+        uses: golangci/golangci-lint-action@v3.4.0
         with:
           # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
-          version: v1.48.0
-          args: --timeout 3m0s --verbose
+          version: v1.51.2
+          args: --timeout 3m0s --verbose --modules-download-mode readonly
```

Comment:
I wonder how this new flag interacts / compares with the later steps in this action (running `go mod tidy` and `git diff --exit-code`)?

Perhaps we don't need to do both?

Reply:
I think you're right that the ` --modules-download-mode readonly` flag should be able to replace `go mod tidy` and `git diff`